### DISCUSSION TITLE
App does not start

### DISCUSSION THREAD

**Discussion by JPVenson** [photoprism/photoprism#3125] [Q&A]
Hey everyone,

I just wanted to setup photoprism in docker with a compose file but it just crashes with an very generic message:

```
photoprism            | running as uid 0
photoprism            | /opt/photoprism/bin/photoprism start
photoprism            | time="2023-01-18T02:52:58Z" level=info msg="config: case-insensitive file system detected"
photoprism            | time="2023-01-18T02:52:58Z" level=error msg="record not found"
photoprism-mariadb-1  | 2023-01-18  2:52:58 3 [Warning] Aborted connection 3 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism exited with code 0
```
It does seem like the connection to the db is fine, so I really dont know whats wrong here.

**Comment by lastzero** [comment]
Seems like you are using an old photoprism version not compatible with mariadb 10.10.

**Comment by FidgetyRat** [comment]
I have a similar issue that happens after I get photoprism up and running a few days. If it's shut down or the VM hosting it crashes, it'll never start again. The only workaround is to wipe the database and start over.

photoprism_1  | /opt/photoprism/bin/photoprism start
photoprism_1  | time="2023-01-21T13:11:55Z" level=info msg="PhotoPrism® needs your support!"
photoprism_1  | time="2023-01-21T13:11:55Z" level=info msg="Visit https://photoprism.app/membership to learn more."
photoprism_1  | time="2023-01-21T13:11:55Z" level=debug msg="config: successfully initialized [2.957002ms]"
mariadb_1     | 2023-01-21 13:11:55 3 [Warning] Aborted connection 3 to db: 'photoprism' user: 'photoprism' host: '<IP>' (Got an error reading communication packets)

Are there any database maintenance strategies to repair or diagnose issues? Otherwise, do we feel that Maria is just too sensitive to potential corruption and should return to sqlite? I need reliability here.

**Reply by lastzero** [reply]
MariaDB is not sensitive and PhotoPrism is configured to drop old connections and start fresh ones every once in a while. We can add yet another config option if you don't like the logs, but this definitely does not crash the database.

**Reply by lastzero** [reply]
My guess is you have memory issues, for example not enough swap? Did you go through our Troubleshooting Checklists yet?

**Reply by FidgetyRat** [reply]
Swap: 
NAME      TYPE      SIZE USED PRIO
/dev/sda2 partition 6.9G   0B   -2

The VM host has 4GB RAM. I'm able to get photoprism running fine, import my whole library (about 9k photos/videos at this time) and it runs fine for a while. If something causes it to shut down, most recently we had a power outage longer than my UPS can handle), it never starts again with the errors above. 

If I completely wipe the database directory and start all over, it works again until the next outage.

The troubleshooting checklist doesn't really have sections for not connecting to the database or at least ones that pertain to what i'm encountering outside the SWAP/Memory section, but its not even using the swap file (100% free). But based on what you said, the inability to start was not caused by the Maria DB warning, but something else not showing in the logs.

I'll try bumping up the virtual RAM temporarily to see if it starts. (Edit, Moved up to 6GB RAM and still won't start with no error logged).

**Reply by lastzero** [reply]
> VM host

If memory is not the problem, I would check the VM hypervisor and storage. I remember MySQL 5 had problems when running on VMware back in the day, but I haven't seen that in ages. Just about all cloud services, including Digital Ocean and AWS, run on virtual machines. They all offer databases and no one loses their data.

**Reply by FidgetyRat** [reply]
What doesn't make sense to me is why dumping the database fixes the problem, which leads me to believe something in the database is corrupt or not right. Unfortunately there is zero logging from photoprism about what it's doing or failed to do when starting up.  I just exits with code 0. 

photoprism_1  | /opt/photoprism/bin/photoprism start
photoprism_1  | time="2023-01-21T14:06:01Z" level=info msg="PhotoPrism® needs your support!"
photoprism_1  | time="2023-01-21T14:06:01Z" level=info msg="Visit https://photoprism.app/membership to learn more."
photoprism_1  | time="2023-01-21T14:06:01Z" level=debug msg="config: successfully initialized [10.467123ms]"
mariadb_1     | 2023-01-21 14:06:01 3 [Warning] Aborted connection 3 to db: 'photoprism' user: 'photoprism' host: '172.19.0.3' (Got an error reading communication packets)
photoprism_photoprism_1 exited with code 0

I'd fully understand a VM issue, or memory, or whatever, but why would starting with a fresh database solve the problem until the next restart?

**Reply by lastzero** [reply]
You could change logging to trace mode to see more.

**Reply by FidgetyRat** [reply]
Thanks, I'll give that a go.

**Reply by FidgetyRat** [reply]
Unfortunately, nothing that interesting in trace.

photoprism_1  | /opt/photoprism/bin/photoprism start
photoprism_1  | time="2023-01-21T14:30:48Z" level=trace msg="config: defaults file /etc/photoprism/defaults.yml does not exist"
photoprism_1  | time="2023-01-21T14:30:48Z" level=debug msg="config: running on 'Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz', 6.2 GB memory detected"
photoprism_1  | time="2023-01-21T14:30:48Z" level=debug msg="settings: loaded from /photoprism/storage/config/settings.yml"
photoprism_1  | time="2023-01-21T14:30:48Z" level=info msg="PhotoPrism® needs your support!"
photoprism_1  | time="2023-01-21T14:30:48Z" level=info msg="Visit https://photoprism.app/membership to learn more."
photoprism_1  | time="2023-01-21T14:30:48Z" level=debug msg="config: successfully initialized [3.152052ms]"
mariadb_1     | 2023-01-21 14:30:48 3 [Warning] Aborted connection 3 to db: 'photoprism' user: 'photoprism' host: '172.19.0.3' (Got an error reading communication packets)
photoprism_photoprism_1 exited with code 0

**Reply by lastzero** [reply]
Are these the latest MariaDB and PhotoPrism versions? Otherwise, my recommendation is to update and also try our development preview to see if it includes changes that fix this.

**Reply by FidgetyRat** [reply]
Looks like I got it working by forcing a MariaDB update from 10.9 to 10.10. Hopefully it sticks this time. Thanks for the help.

**Reply by FidgetyRat** [reply]
And it's back to not working after the first shutdown. Same symptoms.

**Reply by lastzero** [reply]
OK, so this is definitely not normal, but I'm running out of ideas and time to help with this. Let us know if you figure out how to solve the problem and feel to send a pull request with improvements for our Troubleshooting Guide.

**Reply by FidgetyRat** [reply]
Do we have an ability to include or add additional logging to the application? Photoprism is claiming to have been successfully initialized (photoprism_1 | time="2023-01-21T13:11:55Z" level=debug msg="config: successfully initialized [2.957002ms]") and then just Exit 0's immediately after. How can we trace what it's trying to do to help debug what is failing?

I have hanged my docker-compose to use photoprism:preview and mariadb 10.10 which downloaded the two new images which are confirmed running. Were there any recent and required changes to docker-compose or some other config file?

Very strange that it came up the one time after updating the DB to 10.10 for the first time, then reverted back to exit 0.

**Reply by lastzero** [reply]
Compatibility with 10.10 has been fixed. See Release Notes.

**Reply by FidgetyRat** [reply]
Right, but I just updated everything to preview and 10.10 yesterday. Is there something new today I should try?

**Reply by lastzero** [reply]
Not today, it's my birthday and I'm not working.

**Reply by FidgetyRat** [reply]
Just for an update, I've had zero luck getting this application stable. Same symptom, exit 0 with zero useful logging involved even on the preview branch with Maria 10.10.

**Reply by lastzero** [reply]
That's unfortunate. Did you also check Linux kernel security modules and virus scanners?

**Reply by FidgetyRat** [reply]
No virus scanners on this box, purely lan-side and nothing out of the ordinary system-wise. As I mentioned, it did run a few times, and even ran one single time recently when I updated Maria to 10.10, but never started up again after it was cleanly shut down. Same symptoms, Exit 0 with no logging.

**Reply by lastzero** [reply]
You enabled automatic schema upgrades, right? Did they run properly?
https://docs.photoprism.app/getting-started/troubleshooting/mariadb/#version-upgrade

**Reply by FidgetyRat** [reply]
Yeah they did.

I have some additional information that I found interesting. I decided to copy off my directory and rolled the VM back to a clean state, then did full system updates. From that point forward I installed docker and followed the usual procedures. 

Starting with a clean directory worked.
Starting with a clean directory and the previous database worked (but there were no thumbs since sidecar was intentionally empty)
Starting with original directory failed. 
Interesting note: Starting maria db alone then photoprism after results in a failure. Starting both at the same time works.

Does anything at all persist within the docker images themselves or are all mutable files on the external shares? I ask because I also saw a case of failure until I purged and re pulled the images.


Something within the photoprism directory structure must be corrupt preventing it from starting. I decided to start from scratch and re-index the entire archive. At that point im planning to take a disk snapshot as a fail safe.

**Reply by FidgetyRat** [reply]
After importing my entire library again and shutting the containers down, the photoprism container is back to exit 0 after initialization with no useful logging.

**Reply by FidgetyRat** [reply]
A very interesting development. Through trial and error, I'm able to get it to start with the full DB and sidecar only if I have `PHOTOPRISM_DISABLE_CHOWN: "true"  ` in my docker-compose. If set false, it exit 0s on startup.

I'm running as a photoprism user, so there must be some kind of permissions race condition on startup after the initial imports.

**Comment by pr0fsmith** [comment]
I can't even get photoprism start.  I get a similar error:

`Attaching to photoprism-mariadb-1, photoprism-photoprism-1
photoprism-mariadb-1     | 2024-02-03 19:33:26+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.2.2+maria~ubu2204 started.
photoprism-mariadb-1     | 2024-02-03 19:33:26+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
photoprism-mariadb-1     | 2024-02-03 19:33:26+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.2.2+maria~ubu2204 started.
photoprism-mariadb-1     | 2024-02-03 19:33:26+00:00 [Note] [Entrypoint]: MariaDB upgrade not required
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] Starting MariaDB 11.2.2-MariaDB-1:11.2.2+maria~ubu2204 source revision 929532a9426d085111c24c63de9c23cc54382259 as process 1
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Number of transaction pools: 1
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Initializing buffer pool, total size = 512.000MiB, chunk size = 8.000MiB
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Completed initialization of buffer pool
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: End of log at LSN=47629
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Opened 3 undo tablespaces
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: 128 rollback segments in 3 undo tablespaces are active.
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: log sequence number 47629; transaction id 14
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] Plugin 'FEEDBACK' is disabled.
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] Plugin 'wsrep-provider' is disabled.
photoprism-mariadb-1     | 2024-02-03 19:33:26 0 [Note] InnoDB: Buffer pool(s) load completed at 240203 19:33:26
photoprism-mariadb-1     | 2024-02-03 19:33:27 0 [Note] Server socket created on IP: '0.0.0.0'.
photoprism-mariadb-1     | 2024-02-03 19:33:27 0 [Note] Server socket created on IP: '::'.
photoprism-mariadb-1     | 2024-02-03 19:33:27 0 [Note] mariadbd: Event Scheduler: Loaded 0 events
photoprism-mariadb-1     | 2024-02-03 19:33:27 0 [Note] mariadbd: ready for connections.
photoprism-mariadb-1     | Version: '11.2.2-MariaDB-1:11.2.2+maria~ubu2204'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
photoprism-photoprism-1  | started 221105-jammy as root (amd64-prod)
photoprism-photoprism-1  | Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
photoprism-photoprism-1  | https://docs.photoprism.app/getting-started/troubleshooting/
photoprism-photoprism-1  | file umask....: "0002" (u=rwx,g=rwx,o=rx)
photoprism-photoprism-1  | home directory: /photoprism
photoprism-photoprism-1  | assets path...: /opt/photoprism/assets
photoprism-photoprism-1  | storage path..: /photoprism/storage
photoprism-photoprism-1  | config path...: default
photoprism-photoprism-1  | cache path....: default
photoprism-photoprism-1  | backup path...: /photoprism/storage/backups
photoprism-photoprism-1  | import path...: /photoprism/import
photoprism-photoprism-1  | originals path: /photoprism/originals
photoprism-photoprism-1  | running as uid 0
photoprism-photoprism-1  | /opt/photoprism/bin/photoprism start
photoprism-photoprism-1  | time="2024-02-03T19:33:33Z" level=error msg="record not found"
photoprism-mariadb-1     | 2024-02-03 19:33:33 3 [Warning] Aborted connection 3 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0
photoprism-photoprism-1  | started 221105-jammy as root (amd64-prod)
photoprism-photoprism-1  | Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
photoprism-photoprism-1  | https://docs.photoprism.app/getting-started/troubleshooting/
photoprism-photoprism-1  | file umask....: "0002" (u=rwx,g=rwx,o=rx)
photoprism-photoprism-1  | home directory: /photoprism
photoprism-photoprism-1  | assets path...: /opt/photoprism/assets
photoprism-photoprism-1  | storage path..: /photoprism/storage
photoprism-photoprism-1  | config path...: default
photoprism-photoprism-1  | cache path....: default
photoprism-photoprism-1  | backup path...: /photoprism/storage/backups
photoprism-photoprism-1  | import path...: /photoprism/import
photoprism-photoprism-1  | originals path: /photoprism/originals
photoprism-photoprism-1  | running as uid 0
photoprism-photoprism-1  | /opt/photoprism/bin/photoprism start
photoprism-photoprism-1  | time="2024-02-03T19:33:41Z" level=error msg="record not found"
photoprism-mariadb-1     | 2024-02-03 19:33:41 4 [Warning] Aborted connection 4 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0
photoprism-photoprism-1  | file umask....: "0002" (u=rwx,g=rwx,o=rx)
photoprism-photoprism-1  | home directory: /photoprism
photoprism-photoprism-1  | assets path...: /opt/photoprism/assets
photoprism-photoprism-1  | storage path..: /photoprism/storage
photoprism-photoprism-1  | config path...: default
photoprism-photoprism-1  | cache path....: default
photoprism-photoprism-1  | backup path...: /photoprism/storage/backups
photoprism-photoprism-1  | import path...: /photoprism/import
photoprism-photoprism-1  | originals path: /photoprism/originals
photoprism-photoprism-1  | running as uid 0
photoprism-photoprism-1  | /opt/photoprism/bin/photoprism start
photoprism-photoprism-1  | time="2024-02-03T19:33:48Z" level=error msg="record not found"
photoprism-mariadb-1     | 2024-02-03 19:33:48 5 [Warning] Aborted connection 5 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0
photoprism-photoprism-1  | time="2024-02-03T19:33:55Z" level=error msg="record not found"
photoprism-mariadb-1     | 2024-02-03 19:33:55 6 [Warning] Aborted connection 6 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0
photoprism-photoprism-1  | time="2024-02-03T19:34:04Z" level=error msg="record not found"
photoprism-mariadb-1     | 2024-02-03 19:34:04 7 [Warning] Aborted connection 7 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0
photoprism-photoprism-1  | time="2024-02-03T19:34:12Z" level=error msg="record not found"
photoprism-mariadb-1     | 2024-02-03 19:34:12 8 [Warning] Aborted connection 8 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0
photoprism-photoprism-1  | time="2024-02-03T19:34:22Z" level=error msg="record not found"
photoprism-mariadb-1     | 2024-02-03 19:34:22 9 [Warning] Aborted connection 9 to db: 'photoprism' user: 'photoprism' host: '172.22.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0`

Here's my docker compose:
`version: '3.5'

# Example Docker Compose config file for PhotoPrism (Linux / AMD64)
#
# Note:
# - Running PhotoPrism on a server with less than 4 GB of swap space or setting a memory/swap limit can cause unexpected
#   restarts ("crashes"), for example, when the indexer temporarily needs more memory to process large files.
# - If you install PhotoPrism on a public server outside your home network, please always run it behind a secure
#   HTTPS reverse proxy such as Traefik or Caddy. Your files and passwords will otherwise be transmitted
#   in clear text and can be intercepted by anyone, including your provider, hackers, and governments:
#   https://docs.photoprism.app/getting-started/proxies/traefik/
#
# Setup Guides:
# - https://docs.photoprism.app/getting-started/docker-compose/
# - https://docs.photoprism.app/getting-started/raspberry-pi/
# - https://www.photoprism.app/kb/activation
#
# Troubleshooting Checklists:
# - https://docs.photoprism.app/getting-started/troubleshooting/
# - https://docs.photoprism.app/getting-started/troubleshooting/docker/
# - https://docs.photoprism.app/getting-started/troubleshooting/mariadb/
#
# CLI Commands:
# - https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface
#
# All commands may have to be prefixed with "sudo" when not running as root.
# This will point the home directory shortcut ~ to /root in volume mounts.

services:
  photoprism:
    ## Use photoprism/photoprism:preview for testing preview builds:
    image: photoprism/photoprism:latest
    ## Don't enable automatic restarts until PhotoPrism has been properly configured and tested!
    ## If the service gets stuck in a restart loop, this points to a memory, filesystem, network, or database issue:
    ## https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
    restart: unless-stopped
    stop_grace_period: 20s
    depends_on:
      - mariadb
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    ports:
      ## Web server port mapping in the format "Host:Container". To use a different port, change the host port
      ## on the left-hand side and keep the container port, e.g. "80:2342" (for HTTP) or "443:2342 (for HTTPS):
      - "2342:2342"
    environment:
      PHOTOPRISM_ADMIN_USER: "admin"                 # admin login username
      PHOTOPRISM_ADMIN_PASSWORD: "insecure"          # initial admin password (8-72 characters)
      PHOTOPRISM_AUTH_MODE: "password"               # authentication mode (public, password)
      PHOTOPRISM_SITE_URL: "http://127.0.0.1:2342/"  # server URL in the format "http(s)://domain.name(:port)/(path)"
      PHOTOPRISM_DISABLE_TLS: "true"                # disables HTTPS/TLS even if the site URL starts with https:// and a certificate is available
      PHOTOPRISM_DEFAULT_TLS: "true"                 # defaults to a self-signed HTTPS/TLS certificate if no other certificate is available
      PHOTOPRISM_ORIGINALS_LIMIT: 10000               # file size limit for originals in MB (increase for high-res video)
      PHOTOPRISM_HTTP_COMPRESSION: "gzip"            # improves transfer speed and bandwidth utilization (none or gzip)
      PHOTOPRISM_LOG_LEVEL: "debug"                   # log level: trace, debug, info, warning, error, fatal, or panic
      PHOTOPRISM_READONLY: "false"                   # do not modify originals directory (reduced functionality)
      PHOTOPRISM_EXPERIMENTAL: "true"               # enables experimental features
      PHOTOPRISM_DISABLE_CHOWN: "true"              # disables updating storage permissions via chmod and chown on startup
      PHOTOPRISM_DISABLE_WEBDAV: "false"             # disables built-in WebDAV server
      PHOTOPRISM_DISABLE_SETTINGS: "false"           # disables settings UI and API
      PHOTOPRISM_DISABLE_TENSORFLOW: "false"         # disables all features depending on TensorFlow
      PHOTOPRISM_DISABLE_FACES: "false"              # disables face detection and recognition (requires TensorFlow)
      PHOTOPRISM_DISABLE_CLASSIFICATION: "false"     # disables image classification (requires TensorFlow)
      PHOTOPRISM_DISABLE_VECTORS: "false"            # disables vector graphics support
      PHOTOPRISM_DISABLE_RAW: "false"                # disables indexing and conversion of RAW images
      PHOTOPRISM_RAW_PRESETS: "false"                # enables applying user presets when converting RAW images (reduces performance)
      PHOTOPRISM_JPEG_QUALITY: 85                    # a higher value increases the quality and file size of JPEG images and thumbnails (25-100)
      PHOTOPRISM_DETECT_NSFW: "false"                # automatically flags photos as private that MAY be offensive (requires TensorFlow)
      PHOTOPRISM_UPLOAD_NSFW: "true"                 # allows uploads that MAY be offensive (no effect without TensorFlow)
      # PHOTOPRISM_DATABASE_DRIVER: "sqlite"         # SQLite is an embedded database that doesn't require a server
      PHOTOPRISM_DATABASE_DRIVER: "mariadb"            # use MariaDB 10.5+ or MySQL 8+ instead of SQLite for improved performance
      PHOTOPRISM_DATABASE_SERVER: "mariadb:3306"     # MariaDB or MySQL database server (hostname:port)
      PHOTOPRISM_DATABASE_NAME: "photoprism"         # MariaDB or MySQL database schema name
      PHOTOPRISM_DATABASE_USER: "photoprism"         # MariaDB or MySQL database user name
      PHOTOPRISM_DATABASE_PASSWORD: "insecure"       # MariaDB or MySQL database user password
      PHOTOPRISM_SITE_CAPTION: "AI-Powered Photos App"
      PHOTOPRISM_SITE_DESCRIPTION: ""                # meta site description
      PHOTOPRISM_SITE_AUTHOR: ""                     # meta site author
      ## Video Transcoding (https://docs.photoprism.app/getting-started/advanced/transcoding/):
      # PHOTOPRISM_FFMPEG_ENCODER: "software"        # H.264/AVC encoder (software, intel, nvidia, apple, raspberry, or vaapi)
      # PHOTOPRISM_FFMPEG_SIZE: "1920"               # video size limit in pixels (720-7680) (default: 3840)
      # PHOTOPRISM_FFMPEG_BITRATE: "32"              # video bitrate limit in Mbit/s (default: 50)
      ## Run/install on first startup (options: update https gpu tensorflow davfs clitools clean):
      # PHOTOPRISM_INIT: "https gpu tensorflow"
      ## Run as a non-root user after initialization (supported: 0, 33, 50-99, 500-600, and 900-1200):
      # PHOTOPRISM_UID: 1000
      # PHOTOPRISM_GID: 1000
      # PHOTOPRISM_UMASK: 0000
    ## Start as non-root user before initialization (supported: 0, 33, 50-99, 500-600, and 900-1200):
    # user: "1000:1000"
    ## Share hardware devices with FFmpeg and TensorFlow (optional):
    # devices:
    #  - "/dev/dri:/dev/dri"                         # Intel QSV
    #  - "/dev/nvidia0:/dev/nvidia0"                 # Nvidia CUDA
    #  - "/dev/nvidiactl:/dev/nvidiactl"
    #  - "/dev/nvidia-modeset:/dev/nvidia-modeset"
    #  - "/dev/nvidia-nvswitchctl:/dev/nvidia-nvswitchctl"
    #  - "/dev/nvidia-uvm:/dev/nvidia-uvm"
    #  - "/dev/nvidia-uvm-tools:/dev/nvidia-uvm-tools"
    #  - "/dev/video11:/dev/video11"                 # Video4Linux Video Encode Device (h264_v4l2m2m)
    working_dir: "/photoprism" # do not change or remove
    ## Storage Folders: "~" is a shortcut for your home directory, "." for the current directory
    volumes:
      # "/host/folder:/photoprism/folder"                # Example
      - "/srv/photos-videos:/photoprism/originals"               # Original media files (DO NOT REMOVE)
      - "/srv/photos-videos/pictures:/photoprism/originals/pictures"   # *Additional* media folders can be mounted like this
      - "/srv/photos-videos/video:/photoprism/originals/video"
      - "/srv/photos-videos:/photoprism/import"                  # *Optional* base folder from which files can be imported to originals
      - "/srv/dockers/photoprism/storage:/photoprism/storage"  # *Writable* storage folder for cache, database, and sidecar files (DO NOT REMOVE)

  ## Database Server (recommended)
  ## see https://docs.photoprism.app/getting-started/faq/#should-i-use-sqlite-mariadb-or-mysql
  mariadb:
    image: mariadb:11
    ## If MariaDB gets stuck in a restart loop, this points to a memory or filesystem issue:
    ## https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
    restart: unless-stopped
    stop_grace_period: 5s
    security_opt: # see https://github.com/MariaDB/mariadb-docker/issues/434#issuecomment-1136151239
      - seccomp:unconfined
      - apparmor:unconfined
    command: --innodb-buffer-pool-size=512M --transaction-isolation=READ-COMMITTED --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci --max-connections=512 --innodb-rollback-on-timeout=OFF --innodb-lock-wait-timeout=120
    ## Never store database files on an unreliable device such as a USB flash drive, an SD card, or a shared network folder:
    volumes:
      - "./database:/var/lib/mysql" # DO NOT REMOVE
    environment:
      MARIADB_AUTO_UPGRADE: "1"
      MARIADB_INITDB_SKIP_TZINFO: "1"
      MARIADB_DATABASE: "photoprism"
      MARIADB_USER: "photoprism"
      MARIADB_PASSWORD: "insecure"
      MARIADB_ROOT_PASSWORD: "insecure"

  ## Watchtower upgrades services automatically (optional)
  ## see https://docs.photoprism.app/getting-started/updates/#watchtower
  ## activate via "COMPOSE_PROFILES=update docker compose up -d"
  watchtower:
    restart: unless-stopped
    image: containrrr/watchtower
    profiles: ["update"]
    environment:
      WATCHTOWER_CLEANUP: "true"
      WATCHTOWER_POLL_INTERVAL: 7200 # checks for updates every two hours
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      # - "~/.docker/config.json:/config.json" # optional, for authentication if you have a Docker Hub account`

Am I doing something wrong?

**Reply by lastzero** [reply]
Have you already gone through our troubleshooting checklists?
- https://docs.photoprism.app/getting-started/troubleshooting/

**Comment by pr0fsmith** [comment]
Yes I have

Feb 3, 2024 2:45:16 p.m. Michael Mayer ***@***.***>:

**Comment by pr0fsmith** [comment]
It's photoprism that's restarting continuously. MariaDB is running fine.

**Comment by rehan0070** [comment]
Seems like you are using an old photoprism [version](https://happywishes4you.com/) not compatible with mariadb 10.10.
